gemhardware/devices/include/gem/hw/devices/amc/HwGenericAMC.h

```diff
@@ -4,774 +4,702 @@
-         */
-        virtual uint32_t getOptoHybridTriggerLinkCount(uint8_t const& oh, uint8_t const& link, AMCOHLinkCountT const& count);
-
-        /****************************/
-        /** DAQ moudle information **/
-        /****************************/
-        /**
-         * @brief performs a general reset of the GenericAMC
-         */
-        virtual void generalReset();
-
-        /**
-         * @brief performs a reset of the GenericAMC counters
-         */
-        virtual void counterReset();
-
-        /**
-         * @brief performs a reset of the GenericAMC T1 counters
-         */
-        virtual void resetT1Counters();
-
-        /**
-         * @brief performs a reset of the GenericAMC GTX link counters
-         */
-        virtual void linkCounterReset();
-
-        /**
-         * @brief performs a reset of the GenericAMC link
-         * @param link is the link to perform the reset on
-         */
-        virtual void linkReset(uint8_t const& link);
-
-        std::vector<AMCIPBusCounters> m_ipBusCounters; /** for each gtx, IPBus counters */
-
-      protected:
-        uint32_t m_links;    ///< Connected links mask
-        uint32_t m_maxLinks; ///< Maximum supported OptoHybrids as reported by the firmware
-
-        /**
-         * @brief sets the expected board ID string to be matched when reading from the firmware
-         * @param boardID is the expected board ID
-         */
-        void setExpectedBoardID(std::string const& boardID) { m_boardID = boardID; }
-
-        std::string m_boardID;  ///< expected boardID in the firmware
-
-      private:
-        // Do not use default constructor. HwGenericAMC object should only be made using
-        // either connection file method or with a list of URIs and address tables
-        HwGenericAMC();
-
-        // Prevent copying of HwGenericAMC objects
-        HwGenericAMC(const HwGenericAMC& other);      // prevents construction-copy
-        HwGenericAMC& operator=(const HwGenericAMC&); // prevents copying
-
-      };  // class HwGenericAMC
+    namespace amc {
+      class HwGenericAMC : public gem::hw::GEMHwDevice
+        {
+        public:
+
+          /**
+           * Constructor, uses IP resolvable device name
+           */
+          HwGenericAMC(std::string const& amcDevice);
+
+          virtual ~HwGenericAMC();
+
+          /**
+           * Check if one can read/write to the registers on the GenericAMC
+           * @returns true if the GenericAMC is accessible
+           */
+          virtual bool isHwConnected() override;
+
+          /****************************/
+          /** GEM system information **/
+          /****************************/
+          /**
+           * @brief Read the board ID register (should be reimplemented in derived HW, e.g., GLIB)
+           * @returns the AMC board ID as a std::string
+           */
+          virtual std::string getBoardIDString();
```

Comment:
is anything inheriting from this?
if not, should remove all `virtual` keywords, and mark `override` those methods which are inherited from `GEMHwDevice`

Reply:
done